exalearn-ui/src/app/components/new-topic/new-topic.component.ts

```diff
@@ -0,0 +1,79 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { FormBuilder, FormGroup, FormGroupDirective, Validators } from '@angular/forms';
+
+import { NewContentService } from '../../services/new-content.service';
+import { ApiService } from '../../services/api.service';
+import { API_PATH } from '../../constants/api.constants';
+
+@Component({
+	selector: 'app-new-topic',
+	templateUrl: './new-topic.component.html',
+	styleUrls: ['./new-topic.component.scss']
+})
+export class NewTopicComponent implements OnInit {
+	@ViewChild(FormGroupDirective, { static: false })
+	form: FormGroupDirective;
+
+	topicForm: FormGroup;
+	load = false;
+
+	constructor(private fb: FormBuilder, private ncService: NewContentService, private apiServise: ApiService) {
+		this.topicForm = this.fb.group({
+			topics: this.fb.array([
+				this.fb.control('', [
+					Validators.required,
+					Validators.minLength(2),
+					this.ncService.noWhitespaceValidator
+				])
+			])
+		});
+	}
+
+	ngOnInit(): void {}
+
+	get topics(): any {
+		return this.topicForm.get('topics');
+	}
+
+	addTopic(): void {
+		this.topics.push(
+			this.fb.control('', [Validators.required, Validators.minLength(2), this.ncService.noWhitespaceValidator])
+		);
+	}
+
+	deleteTopic(index: number): void {
+		this.topics.removeAt(index);
+	}
+
+	resetForm(): void {
+		this.form.resetForm();
+
+		while (this.topics.length !== 1) {
+			this.deleteTopic(0);
+		}
+	}
+
+	onSubmit(): void {
+		this.load = true;
+		console.log('send to server:', JSON.stringify(this.topicForm.value));
+
+		setTimeout(() => {
+			this.resetForm();
+			this.load = false;
+		}, 2000);
+
+		//TODO: delete code above when server is ready
+		// and uncomment code lower
+
+		/* 
+		this.load = true;
```

Comment:
Please remove comments

Reply:
I marked this comment like TODO. I will remove my code which located above and uncomment code below. Because code below is ready to send data to server i commented it. I made it with one purpose - so the rest of us can test this component and identify some errors, bugs and wishes.